deploy/example_clusters/cr-cluster-tensorflow-notebook.yaml

```diff
@@ -10,15 +10,12 @@ spec:
       requests:
         memory: "4Gi"
         cpu: "2"
+        nvidia.com/gpu: "1"        
       limits:
         memory: "4Gi"
         cpu: "2"
-  - id: worker
-    members: 1
-    resources:
-      requests:
-        memory: "4Gi"
-        cpu: "2"
-      limits:
-        memory: "4Gi"
-        cpu: "2"
+        nvidia.com/gpu: "1"        
+    env:
+      - name : "LD_LIBRARY_PATH"
+        value: "/usr/local/nvidia/lib64"
```

Comment:
Hmm. If the user is required to enter this in the kdcluster spec each time, it's too brittle for usability IMO. Does the official image really not work without this?

Some other solutions:

- Modifying the image to include this in /etc/environment or some other relevant profile. Obviously this is a bummer if you want to use the unmodified image.
- Adding a startscript package that does that modification at runtime. Should work fine but does require the startscript package residing on some available http server. A bummer if we want our example kdapps to be self-contained.
- New feature in KD to allow specifying env vars in the role definition of the KubeDirectorApp. Final set provided to the pod would be the env vars from the kdapp unified with (overridden by) any env vars from the kdcluster.

Let me know your thoughts.

Reply:
#2 is doable by having start-script baked into the image. Although #3 is ideal, might not be possible for 5.0 if we want this to be shipped

Reply:
hmm, actually it will 1st not 2nd then.

Reply:
First we should determine if it really is required, i.e. if the official image is broken without specifying this. I would be surprised-ish, but stranger things have happened.

Reply:
This is what i had before

https://github.com/bluek8s/kubedirector/blob/v0.3.0/deploy/example_catalog/cr-app-tensorflowgpu.json

This just worked without any changes

Reply:
Yes, it works even without the environment variable. I'll remove it.